Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM jupyter/base-notebook:latest
+
+LABEL maintainer="Fabian Brandt-Tumescheit <brandtfa@hu-berlin.de>"
+
+USER root
+RUN mkdir /var/lib/apt/lists/partial && \
+    apt-get update && \
+    apt-get install -y  --no-install-recommends cmake git build-essential && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+USER 1000
+RUN pip install --upgrade pip setuptools cython
+RUN pip install powerlaw
+RUN pip install sklearn
+RUN pip install seaborn
+RUN pip install networkit
+RUN pip install pandas
+RUN pip install tabulate
+RUN pip install matplotlib
+RUN pip install networkx
+RUN pip install ipycytoscape
+
+RUN git clone https://github.com/networkit/networkit.git
+WORKDIR networkit
+RUN git checkout 8.0
+RUN mv notebooks ${HOME}/
+RUN mv input ${HOME}/
+RUN rm ${HOME}/notebooks/test_notebooks.py
+WORKDIR ..
+RUN rm -rf networkit
+
+RUN conda install -y nodejs
```

Comment:
Why do we install nodejs from conda, not from apt?

Reply:
It is based on recommendation: This is because the complete jupyter environment (the image is built upon it) is part of a conda environment. This also capsulates user permissions.